File: backend/app/schemas/sensor.py
"""Sensor data contracts."""

from datetime import datetime

from pydantic import Field

from app.models.sensor import SensorSourceClass, SensorStatus
from app.schemas._operational import OperationalSchema


class SensorSchema(OperationalSchema):
    id: str | None = None
    name: str = Field(min_length=1, max_length=200)
    source_type: str = Field(min_length=1, max_length=64)
    source_class: SensorSourceClass
    status: SensorStatus = SensorStatus.REGISTERED
    configuration_version: int = Field(default=1, ge=1)
    configuration_metadata: dict = Field(default_factory=dict)
    created_at: datetime | None = None
    updated_at: datetime | None = None

File: apps/operator/src/types/sensor.ts
export type SensorStatus = 'ACTIVE' | 'INACTIVE' | 'MAINTENANCE' | 'DEGRADED';
export type SensorSourceClass = 'REAL' | 'SIMULATION' | 'REPLAY';

export interface Sensor {
  id: string;
  name: string;
  source_type: string;
  source_class: SensorSourceClass;
  status: SensorStatus;
  configuration_version: number;
  configuration_metadata?: {
    latitude?: number;
    longitude?: number;
    altitude?: number;
    range_meters?: number;
    detection_probability?: number;
    [key: string]: unknown;
  };
  created_at: string;
  updated_at: string;
}

export interface SensorListResponse {
  items: Sensor[];
  total: number;
  limit: number;
  offset: number;
}
